sale_request/models/sale_request.py

```diff
@@ -269,13 +270,121 @@ def name_get(self):
     def _onchange_product_id(self):
         if self.product_id:
             sol_obj = self.env['sale.order.line']
+            self._onchange_product_id_check_availability()
+
             self.update({
                 'product_uom_id': self.product_id.uom_id,
                 'description': (
                     sol_obj.get_sale_order_line_multiline_description_sale(
                         self.product_id)),
             })
 
+    @api.onchange('product_id')
+    def _onchange_product_id_uom_check_availability(self):
+        self._onchange_product_id_check_availability()
+
+    # This is a copy from original method that belong to sale.order.line
+    # this method is adapted to sale.request.line model
+    @api.onchange('product_qty')
+    def _onchange_product_id_check_availability(self):
+        if not self.product_id or not self.product_qty:
+            return {}
+        if self.product_id.type == 'product':
+            precision = self.env['decimal.precision'].precision_get(
+                'Product Unit of Measure')
+            product = self.product_id.with_context(
+                warehouse=self.request_id.warehouse_id.id,
+                lang=self.request_id.partner_id.lang or
+                self.env.user.lang or 'en_US'
+            )
+            product_qty = self.product_qty
+            # product_qty = self.product_uom._compute_quantity(self.product_uom_qty, self.product_id.uom_id)
```

Comment:
why you comment this?

Reply:
I forget remove that line

Reply:
Don't remove this line, it's important to verify the quantity in the product default UoM